File: backend/src/main/java/com/team05/petmeeting/domain/feed/dto/AdoptedAnimalRes.java
package com.team05.petmeeting.domain.feed.dto;

import com.team05.petmeeting.domain.animal.entity.Animal;

public record AdoptedAnimalRes(
        Long animalId,
        String noticeNo,
        String upKindNm,    // 개/고양이
        String kindFullNm,  // 품종
        String imageUrl
) {
    public static AdoptedAnimalRes from(Animal animal) {
        return new AdoptedAnimalRes(
                animal.getId(),
                animal.getNoticeNo(),
                animal.getUpKindNm(),
                animal.getKindFullNm(),
                animal.getPopfile1()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/feed/dto/AdoptedAnimalRes.kt
package com.team05.petmeeting.domain.feed.dto

import com.team05.petmeeting.domain.animal.entity.Animal

data class AdoptedAnimalRes(
    val animalId: Long?,
    val noticeNo: String?,
    val upKindNm: String?,  // 개/고양이
    val kindFullNm: String?,  // 품종
    val imageUrl: String?
) {
    companion object {
        fun from(animal: Animal): AdoptedAnimalRes {
            return AdoptedAnimalRes(
                animal.id,
                animal.noticeNo,
                animal.upKindNm,
                animal.kindFullNm,
                animal.popfile1
            )
        }
    }
}


